aea/cli/config.py

```diff
@@ -66,79 +149,20 @@ def convert(self, value, param, ctx):
             'skills.my_skill.an_attribute_name'
             'vendor.author.[protocols|connections|skills].package_name.attribute_name
         """
-        parts = value.split(".")
-
-        root = parts[0]
-        if root not in ALLOWED_PATH_ROOTS:
-            self.fail(
-                "The root of the dotted path must be one of: {}".format(
-                    ALLOWED_PATH_ROOTS
-                )
-            )
-
-        if (
-            len(parts) < 1
-            or parts[0] == "agent"
-            and len(parts) < 2
-            or parts[0] == "vendor"
-            and len(parts) < 5
-            or parts[0] != "agent"
-            and len(parts) < 3
-        ):
-            self.fail(
-                "The path is too short. Please specify a path up to an attribute name."
-            )
-
-        # if the root is 'agent', stop.
-        if root == "agent":
-            resource_type_plural = "agents"
-            path_to_resource_configuration = DEFAULT_AEA_CONFIG_FILE
-            json_path = parts[1:]
-        elif root == "vendor":
-            resource_author = parts[1]
-            resource_type_plural = parts[2]
-            resource_name = parts[3]
-            path_to_resource_directory = (
-                Path(".")
-                / "vendor"
-                / resource_author
-                / resource_type_plural
-                / resource_name
-            )
-            path_to_resource_configuration = (
-                path_to_resource_directory
-                / RESOURCE_TYPE_TO_CONFIG_FILE[resource_type_plural]
-            )
-            json_path = parts[4:]
-            if not path_to_resource_directory.exists():
-                self.fail(
-                    "Resource vendor/{}/{}/{} does not exist.".format(
-                        resource_author, resource_type_plural, resource_name
-                    )
-                )
+        try:
+            (
+                json_path,
+                path_to_resource_configuration,
+                config_loader,
+            ) = handle_dotted_path(value)
+        except Exception as e:
+            self.fail(str(e))
         else:
```

Comment:
why the else?

Reply:
To put in `try` statement only the call we catch exception from. And `else` block includes the correct result handling.